src/components/RegistrationForm.tsx

```diff
@@ -0,0 +1,60 @@
+import { useState, ChangeEventHandler, ChangeEvent } from "react";
+import { db as firebaseDB } from "../services/firebase-client";
+import { sha256 } from "js-sha256";
+
+export const RegistrationForm = ({ onRegistrationDone = (() => {}) }) => {
+  
+  const [username, setUsername] = useState("");
+  const [redemptionType, setRedemptionType] = useState("");
+
+  const InputChangeHandlers: { [key: string]: ChangeEventHandler<HTMLInputElement> } = {
+    username: (e: ChangeEvent<HTMLInputElement>) => {
+      setUsername(e.target.value);
+    },
+    redemptionType: (e: ChangeEvent<HTMLInputElement>) => {
+      setRedemptionType(e.target.value);
+    },
+  };
+  const createInputChangeHandler = ( name: string ): ChangeEventHandler<HTMLInputElement> => {
+    const handleInputChange = InputChangeHandlers[name] || ((event: ChangeEvent<HTMLInputElement>) => {});
+
+    return handleInputChange;
+  };
+  const registerUser = async () => {
+    const createDate = new Date();
+    const redemptionString = `${username}_${redemptionType}-${createDate}`;
+    const redemptionCode = sha256(redemptionString)
+
+    try {
+      await firebaseDB.append("users", {
+        username,
+        redemptionType,
+        redemptionCode,
+        createDate,
+      });
+
+      onRegistrationDone();
+    } catch (e) {
+      console.error(e)
```

Comment:
How would you handle it? Add `# TODO:` tag if you intented to handle
otherwise, just let it crashes